File: chapter10/typescript/EuropeanCoffeeMachine.ts
import { EuropeanAppliance } from "./EuropeanAppliance";


/**
 * EuropeanCoffeeMachine implements run method which make sure input voltage can run European Coffee Machine
 *
 * @Author: Bridget
 */
export class EuropeanCoffeeMachine extends EuropeanAppliance {

    /**
     * check voltage input is 220 and can run European Coffee Machine to Brew Coffee
     *
     * @param volt
     */
     public run(volt: number): void{
         let isVoltCompatible = this.checkVoltageCompatible(volt);
         console.log("EuropeanCoffeeMachine: Input Voltage is ", volt);
         if (!isVoltCompatible) {
            console.log("Voltage - (" + volt + ") is not supported to use European Coffee Machine");
         } else {
            console.log("Brew Coffee!");
         }
     }
}

File: chapter10/java/designpattern/src/main/java/com/designpattern/EuropeanCoffeeMachine.java
package com.designpattern;

/**
 * EuropeanCoffeeMachine implements run method which make sure input voltage can run European Coffee Machine
 *
 * @Author: Bridget
 */

public class EuropeanCoffeeMachine implements EuropeanAppliance {

    /**
     * check voltage input is 220 and can run European Coffee Machine to Brew Coffee
     *
     * @param volt
     */
    @Override
    public void run(int volt) {
        boolean isVoltCompatible = checkVoltageCompatible(volt);
        System.out.println("EuropeanCoffeeMachine: Input Voltage is " + volt);
        if (!isVoltCompatible) {
            System.out.println("Voltage - (" + volt + ") is not supported to use European Coffee Machine");
        } else {
            System.out.println("Brew Coffee!");
        }
    }
}
